lib/bindings/http/emitter_structured.js

```diff
@@ -1,30 +1,23 @@
 var axios = require("axios");
 
 const Constants = require("./constants.js");
+const defaults = {};
+defaults[Constants.HEADERS] = {};
+defaults[Constants.HEADERS][Constants.HEADER_CONTENT_TYPE] = Constants.DEFAULT_CE_CONTENT_TYPE;
 
 function StructuredHTTPEmitter(configuration){
-  this.config = JSON.parse(JSON.stringify(configuration));
-
-  this.config[Constants.HEADERS] =
-    (!this.config[Constants.HEADERS]
-      ? {}
-      : this.config[Constants.HEADERS]);
-
-  if(!this.config[Constants.HEADERS][Constants.HEADER_CONTENT_TYPE]){
-    this.config[Constants.HEADERS][Constants.HEADER_CONTENT_TYPE] =
-      Constants.DEFAULT_CE_CONTENT_TYPE;
-  }
+  this.config = Object.assign({}, defaults, configuration);
 }
 
-StructuredHTTPEmitter.prototype.emit = function(cloudevent) {
-  // Create new request object
-  var _config = JSON.parse(JSON.stringify(this.config));
-
+StructuredHTTPEmitter.prototype.emit = function (cloudevent) {
   // Set the cloudevent payload
-  _config[Constants.DATA_ATTRIBUTE] = cloudevent.format();
+  this.config[Constants.DATA_ATTRIBUTE] = cloudevent.format();
 
   // Return the Promise
-  return axios.request(_config);
+  return axios.request(this.config).then(response => {
+    delete this.config[Constants.DATA_ATTRIBUTE];
+    return response;
+  });
```

Comment:
This is very confusing.

Why don't we just do this?

```js
const res = await axios.request(this.config);
delete this.config[Constants.DATA_ATTRIBUTE];
return res;
```

The outer function needs to be async of course.

Reply:
This change introduces a race condition.

Reply:
No, where's the race? That's not how Node executes.

Reply:
I vote for `then` notation.

Reply:
@grant try this in your browser https://gist.github.com/lance/ff1bf6d2da191ad164257349486c4c95